package.json

```
@@ -250,8 +250,8 @@
 		"test-unit:debug": "wp-scripts --inspect-brk test-unit-js --runInBand --no-cache --verbose --config test/unit/jest.config.js ",
 		"test-unit:update": "npm run test-unit -- --updateSnapshot",
 		"test-unit:watch": "npm run test-unit -- --watch",
-		"test-unit-php": "wp-env run phpunit 'phpunit -c /var/www/html/wp-content/plugins/gutenberg/phpunit.xml.dist --verbose'",
-		"test-unit-php-multisite": "wp-env run phpunit 'WP_MULTISITE=1 phpunit -c /var/www/html/wp-content/plugins/gutenberg/phpunit/multisite.xml --verbose'",
+		"test-unit-php": "wp-env start && wp-env run phpunit 'phpunit -c /var/www/html/wp-content/plugins/gutenberg/phpunit.xml.dist --verbose'",
+		"test-unit-php-multisite": "wp-env start && wp-env run phpunit 'WP_MULTISITE=1 phpunit -c /var/www/html/wp-content/plugins/gutenberg/phpunit/multisite.xml --verbose'",
```

Aside, should we also add

```suggestion
		"pretest-unit-php": "wp-env run composer install -- --no-interaction",
		"test-unit-php": "wp-env start && wp-env run phpunit 'phpunit -c /var/www/html/wp-content/plugins/gutenberg/phpunit.xml.dist --verbose'",
		"pretest-unit-php-multisite": "wp-env run composer install -- --no-interaction",
		"test-unit-php-multisite": "wp-env start && wp-env run phpunit 'WP_MULTISITE=1 phpunit -c /var/www/html/wp-content/plugins/gutenberg/phpunit/multisite.xml --verbose'",
```

? I know that @Copons was running into that issue when first trying to `npm run test-unit-php`.

---

I'm not familiar with this kind of things, but yes, I can confirm that `npm run test-unit-php` failed for me when spinning up the Docker images, and it was fixed (as suggested by @ockham) after running `composter install`.

---

The problem with doing that is when you run `test-php`, then it's going to run `composer install` 3 times. May not be a huge issue, but it will add some additional overhead.

---

Indeed :confused: Is there maybe some CLI option to make `composer install` smart enough to only install stuff if something has changed?

---

It will detect and not install anything if there are no changes, but it still has to check. I do think it regenerates the autoload files by default unless you pass `--no-autoloader`. It's fairly quick, but depends on how many dependencies there are.

---

True.. Could we just remove the lint check from `test-php`?

---

If we don't end up executing `composer install` automatically, would it be worthwhile to add a note under `## PHP Testing` in `docs/contributors/testing-overview.md` about running composer install before `npm run test-unit-php`?

---

yep!